Content.Server/CriminalRecords/Systems/GeneralCriminalRecordConsoleSystem.cs

```diff
@@ -0,0 +1,252 @@
+
+        if (!_criminalRecordsSystem.TryArrest(station!.Value, component.ActiveKey!.Value, out var status, msg.Reason))
+            return;
+
+        if (msg.Reason != string.Empty)
+        {
+            var messages = new Dictionary<SecurityStatus, string>()
+            {
+                { SecurityStatus.Detained, Loc.GetString("general-criminal-record-console-detained-with-reason", ("name", msg.Name), ("reason", msg.Reason)!, ("goodguyname", Name(msg.Session.AttachedEntity.Value))) },
+                { SecurityStatus.None, Loc.GetString("general-criminal-record-console-undetained-with-reason", ("name", msg.Name), ("reason", msg.Reason)!, ("goodguyname", Name(msg.Session.AttachedEntity.Value))) }
+            };
+
+            _radioSystem.SendRadioMessage(uid, messages[status],
+                _prototypeManager.Index<RadioChannelPrototype>("Security"), uid);
+        }
+
+        else if (msg.Reason == string.Empty)
+        {
+            var messages = new Dictionary<SecurityStatus, string>()
+            {
+                { SecurityStatus.Detained, Loc.GetString("general-criminal-record-console-detained-without-reason", ("name", msg.Name), ("goodguyname", Name(msg.Session.AttachedEntity.Value))) },
+                { SecurityStatus.None, Loc.GetString("general-criminal-record-console-undetained-without-reason", ("name", msg.Name), ("goodguyname", Name(msg.Session.AttachedEntity.Value))) }
+            };
+
+            _radioSystem.SendRadioMessage(uid, messages[status],
+                _prototypeManager.Index<RadioChannelPrototype>("Security"), uid);
+        }
+
+        UpdateUserInterface(uid, component);
+    }
+
+    private void OnStatusSelected(EntityUid uid, GeneralCriminalRecordConsoleComponent component,
+        CriminalStatusOptionButtonSelected msg)
+    {
+        if (msg.Session.AttachedEntity is not {Valid: true} mob) return;
+        if (!CanUse(mob, uid))
+        {
+            _popupSystem.PopupEntity(Loc.GetString("general-criminal-record-permission-denied"), uid, msg.Session);
+            return;
+        }
+
+        var station = _stationSystem.GetOwningStation(msg.Session.AttachedEntity!.Value);
+
+        if (!_criminalRecordsSystem.TryChangeStatus(station!.Value, component.ActiveKey!.Value, msg.Status,
+                out var status, msg.Reason))
+            return;
+
+        if (msg.Reason != string.Empty)
+        {
+            var messages = new Dictionary<SecurityStatus, string>()
+            {
+                { SecurityStatus.Wanted, Loc.GetString("general-criminal-record-console-wanted-with-reason", ("name", msg.Name), ("reason", msg.Reason)!, ("goodguyname", Name(msg.Session.AttachedEntity.Value))) },
+                { SecurityStatus.None, Loc.GetString("general-criminal-record-console-not-wanted-with-reason", ("name", msg.Name), ("reason", msg.Reason)!, ("goodguyname", Name(msg.Session.AttachedEntity.Value))) }
+            };
+
```

Comment:
As elsewhere this looks a little too copy-pastey with the above method + questionable null suppression.